### DISCUSSION TITLE
Property 'readContract' does not exist on type

### DISCUSSION THREAD

**Discussion by shahzad73** [wevm/viem#1211] [General]
I have following error 

TS2339: Property 'readContract' does not exist on type '{ account: undefined; batch?: { multicall?: boolean | { batchSize?: number | undefined; wait?: number | undefined; } | undefined; } | undefined; cacheTime: number; chain: Assign<{ readonly id: 11155111; ... 6 more ...; readonly testnet: true; }, ChainConfig<...>>; ... 10 more ...; extend: () => const; }'.
    408 |       const fetchTotalSupply = async () => {
    409 |
  > 410 |         const data = await publicClient.readContract({
        |                                         ^^^^^^^^^^^^
    411 |           address: '0x6b1792a04fcf47e9a2e8ee6c8d455e9e6ce8c5bb',
    412 |           abi: [{
    413 |             "inputs": [],

and the code is 

        const data = await publicClient.readContract({
          address: '0x6b1792a04fcf47e9a2e8ee6c8d455e9e6ce8c5bb',
          abi: [{
            "inputs": [],
            "name": "totalSupply",
            "outputs": [
              {
                "internalType": "uint256",
                "name": "",
                "type": "uint256"
              }
            ],
            "stateMutability": "view",
            "type": "function",
          }],
          functionName: 'totalSupply'
        })

        setSupplyTotal( data.toString() );

Code does execute and fetches totalSupply    BUT    it also raises error          any ideas what might be wronge

**Comment by jxom** [comment]
I can't seem to reproduce the type error [here](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgYygUwIYzQBQK4BGANsMgMIloB2MANHABYwxhwC+cAZlBCHAOQA3YGhD8AUOOQQqAZ3hhCJcpRpwAvCnRZcS0hRE0AFAnFw4MKBjmRYALkbMwRgJTi2bqTPlwAJlgwNOAwAdwxgBT0VQxgAOm1fMhlLDGQYEzNg3190WVkHACIABgAPADYCAEYAdgBOACYMIoAWTmROZuq0Wox6tAAONDQy5H7fZoBWCe7h5AHkCYICAtpMjAJgBwBtTPNTcwO4YCpFGHy4LYBdVcPzKgwQNEKYCBgMIgBlPDAwIgBPFa7A4QPAsUHnHa3A77KEHY7YKD3IgAFT+YCecAKeHh9QmZUBsIO90ehQJhIsaIxWJxeLJULYN1u1yB5nkOgAsqD1sASDA-oVhGgQnSDnz0YVOHgqGlgDIRQzMszzJLpTBZVQAHIPKkvN6fb6-AGrDwAbnEQA). Can you try reproduce in a minimal environment and share it here?

**Reply by shahzad73** [reply]
yee it's a strange error,   because i am copying code from viem.sh       what do you mean by minimal environment ?

**Reply by shahzad73** [reply]
and strange thing is the code works     it does read contract and bring data that is shown on the interface    but then this error is shown.   i just have to close the overlay error message

**Comment by hejtao** [comment]
Upgrading the typescript version helps me solve the problem.
